{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5f43e3b9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting socketio\n",
      "  Using cached socketio-0.2.1-py3-none-any.whl\n",
      "Requirement already satisfied: setuptools==3.3 in e:\\anaconda3\\lib\\site-packages (from socketio) (3.3)\n",
      "Collecting netifaces==0.10.6\n",
      "  Using cached netifaces-0.10.6.tar.gz (25 kB)\n",
      "Collecting socketio\n",
      "  Using cached socketio-0.1.9.tar.gz (4.5 kB)\n",
      "  Using cached socketio-0.1.7.tar.gz (4.5 kB)\n",
      "  Using cached socketio-0.1.6.zip (6.9 kB)\n",
      "  Using cached socketio-0.1.5.zip (6.9 kB)\n",
      "  Using cached socketio-0.1.4.zip (6.8 kB)\n",
      "  Using cached socketio-0.1.3.tar.gz (4.5 kB)\n",
      "  Using cached socketio-0.1.2.tar.gz (4.2 kB)\n",
      "  Using cached socketio-0.1.1.tar.gz (4.3 kB)\n",
      "  Using cached socketio-0.1.0.tar.gz (4.3 kB)\n",
      "  Using cached socketio-0.0.9.tar.gz (4.3 kB)\n",
      "  Using cached socketio-0.0.8.tar.gz (4.3 kB)\n",
      "  Using cached socketio-0.0.7.tar.gz (4.2 kB)\n",
      "  Using cached socketio-0.0.6.tar.gz (4.2 kB)\n",
      "  Using cached socketio-0.0.5.tar.gz (4.2 kB)\n",
      "  Using cached socketio-0.0.4.tar.gz (4.2 kB)\n",
      "  Using cached socketio-0.0.3.tar.gz (4.1 kB)\n",
      "  Using cached socketio-0.0.2.tar.gz (4.1 kB)\n",
      "  Using cached socketio-0.0.1.tar.gz (3.5 kB)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\netifaces_1a77dc2b2afa492fb739ed0dc73020ce\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\netifaces_1a77dc2b2afa492fb739ed0dc73020ce\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-ycqqv2rz'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\netifaces_1a77dc2b2afa492fb739ed0dc73020ce\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/72/01/ba076082628901bca750bf53b322a8ff10c1d757dc29196a8e6082711c9d/netifaces-0.10.6.tar.gz#sha256=0c4da523f36d36f1ef92ee183f2512f3ceb9a9d2a45f7d19cda5a42c6689ebe0 (from https://pypi.org/simple/netifaces/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_6a9881dd3eb34531b1f795176c09b510\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_6a9881dd3eb34531b1f795176c09b510\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-t3giog23'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_6a9881dd3eb34531b1f795176c09b510\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/b8/54/d97b36f1915673a08b434019a3f26fb4c681282ae33c0c3c6f70938f039c/socketio-0.1.9.tar.gz#sha256=01be7d6d3cb190e0222e82f03c073070d869b921b88c3440d809a012ceb3d788 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_d0c76aeac4894cd4baf7af9f61e985e5\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_d0c76aeac4894cd4baf7af9f61e985e5\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-npclbpmn'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_d0c76aeac4894cd4baf7af9f61e985e5\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/62/20/c25be4eab68780f7b55e49511dcd3c71b31f1af597595cf6b2584b3acb14/socketio-0.1.7.tar.gz#sha256=2bb2ef48c481df4595465f9ea5822087f0156920964b292009e63dfeb02ec746 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_3299595f1268468ca025c49616996762\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_3299595f1268468ca025c49616996762\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-xz1qks3t'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_3299595f1268468ca025c49616996762\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/96/81/57a84d67ef2d8cfd4f5699d938cccf5be580026dd78356ba66276d5699a9/socketio-0.1.6.zip#sha256=57ee7291b8515971ccf466a90e39dacbda929ef64fe69984f0daca968cbb8a37 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_98682330605b4d13a7aaf465f636857c\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_98682330605b4d13a7aaf465f636857c\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-ueoxuc0j'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_98682330605b4d13a7aaf465f636857c\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/10/26/0f0c2ca2bb994023deba3fa54619d068bb11eab2a02d44a49d90a0f57407/socketio-0.1.5.zip#sha256=7158773fd71017b4dc8add962ec409af69091cca4e4255347d407159ff995a1d (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_c00d2612d6bc494b804290e9c7d23078\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_c00d2612d6bc494b804290e9c7d23078\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-xd795c2n'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_c00d2612d6bc494b804290e9c7d23078\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/7e/c9/44c96fdea0802845ac1a11f5c2b141c9995fb6bc88c66319faeecd3f8624/socketio-0.1.4.zip#sha256=a168c788768436434f3de91f919af0d9e7fb2c7ffbe1d3e359f0e6d9e71efd0b (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_15ea30f462af470486963bf9fa98dba3\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_15ea30f462af470486963bf9fa98dba3\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-73_m1riz'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_15ea30f462af470486963bf9fa98dba3\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/32/fb/8667be5433aa2f54c8111d37f75425789ef7425c1c8aaa65b99c36c460de/socketio-0.1.3.tar.gz#sha256=c43e5b73b9cccb9ff7b05933415c54d302176572d97a03736c86c2191178a9c5 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_f5611367c3fd4c2197d48e734584eff7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_f5611367c3fd4c2197d48e734584eff7\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-v4lb9iq_'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_f5611367c3fd4c2197d48e734584eff7\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/b8/20/b0e18abbe52252497fa05fd1d2eb89e766bc98ebc442d9a6bfdba9e9d164/socketio-0.1.2.tar.gz#sha256=9f747eeb8d80f91bfbc57bb4d6db8d65948aa259a5cef4e2a76ff2e3e59b5424 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_54c5dcfb80254322b1c6645fa891f351\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_54c5dcfb80254322b1c6645fa891f351\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-of4v48wi'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_54c5dcfb80254322b1c6645fa891f351\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/51/8f/52b3f1cd26b7682c2606916f3bb18b26c409605b68773128cf5e5a728efc/socketio-0.1.1.tar.gz#sha256=02b551453ef0ea5f5e3e164eabdd91f4cec561c919d38a2556007bd470ba4ed2 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_f8a9bc90e32448c09c6967731e932d36\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_f8a9bc90e32448c09c6967731e932d36\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-vp2_et0w'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_f8a9bc90e32448c09c6967731e932d36\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/49/91/4584341e0e86a90e404066c53b3faab71ebc3ef112c4fd4cc118a41a2e99/socketio-0.1.0.tar.gz#sha256=58902a8f29617bf1407bd2351beeb14c149691e6dc8b967d86180fe1dcc76931 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_f3b84c98fa114b8aa38429e1df1900e0\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_f3b84c98fa114b8aa38429e1df1900e0\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-5ik7cg3u'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_f3b84c98fa114b8aa38429e1df1900e0\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/33/4b/6650a048e945cc63de7aa68c9f37029acdfd237549efaf2d245e2cff5b6c/socketio-0.0.9.tar.gz#sha256=4286dd89dd1f840039699488798810a054dbcbdf82b3636e6792abeaa0efccfd (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_7fb31ceabb374cc586551cdf68d5e1d1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_7fb31ceabb374cc586551cdf68d5e1d1\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-4x2z8o98'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_7fb31ceabb374cc586551cdf68d5e1d1\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/b3/44/eff9c853fef62ec2088107ee526d00d36c97e9fce3ddd74f989e90d5f781/socketio-0.0.8.tar.gz#sha256=6331258355c1f46318e101b2b32d63dfad4d952496279b0cfe1cf70a2d5c80af (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_a9237b068c6f4059a4b8d3e87e810acf\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_a9237b068c6f4059a4b8d3e87e810acf\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-ouzdi_p2'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_a9237b068c6f4059a4b8d3e87e810acf\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/72/95/7fa1739c9fd8ef671ebf1bbe7ae55cd1a9be17368a19a56731851311c533/socketio-0.0.7.tar.gz#sha256=56af0abf5b5c0b27129adbda5da19b95448aeb4da958a9e6c081fa26f34f72d1 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_aed4b7cb9f9d4d0b99a1a5dfe7c539ec\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_aed4b7cb9f9d4d0b99a1a5dfe7c539ec\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-tg4i9xso'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_aed4b7cb9f9d4d0b99a1a5dfe7c539ec\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/c3/47/ba89f48d48150e9f0ccf18909f8eabbf6640ab996214cff05e999ec04f41/socketio-0.0.6.tar.gz#sha256=a54e7a469dd888b684badade6508b32489df6f1c5e0648e3d0207546fe1f10d5 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_e255d7b2161a467f95b3f13c2a663826\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_e255d7b2161a467f95b3f13c2a663826\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-3idv_rg4'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_e255d7b2161a467f95b3f13c2a663826\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/3d/ed/8a50081102b7c1692d1f97e08e292ff1a9f0ab2c52bfd61c53276ee5a007/socketio-0.0.5.tar.gz#sha256=3a651f6abf2370857579aa70f52b5d0ab2348ab31686140ebce688ca43df9861 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_a45e0304702940d2b11948924c590609\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_a45e0304702940d2b11948924c590609\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-vyreeg54'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_a45e0304702940d2b11948924c590609\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/dd/4f/e4df45925f0bb4d54b96529ac22104db32950a167f7ac5e9ebbb1360f7f1/socketio-0.0.4.tar.gz#sha256=1318839b366fcae35f83358dfee74c047947ffc676f853dc442d270fc1411a3d (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_9d1462b2feb046dca8cd02cc8534885f\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_9d1462b2feb046dca8cd02cc8534885f\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-7z04wz9m'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_9d1462b2feb046dca8cd02cc8534885f\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/37/48/3740f312cca1fef23349e227a85bee7b1aaed51ca55d37d8076a8d7359e9/socketio-0.0.3.tar.gz#sha256=2c39f0bfc7234869bbb97370396e0012a39178ad2a6d018f2cfb2246f86cd977 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_a709f8f0527b49a49325d950ba7a34a4\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_a709f8f0527b49a49325d950ba7a34a4\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-_86c3w7s'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_a709f8f0527b49a49325d950ba7a34a4\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/39/5a/c51fbfdc7c53a000a8180b544afacab4e5e7ad8ca1e68cbaf7c21842ce65/socketio-0.0.2.tar.gz#sha256=c192df5b43657017d8a555c49f7489bf035fa7b2a3cfb81f6adb6d52c9ca1216 (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'E:\\anaconda3\\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_697229108dcc4cf39a5262e114ca36cd\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\HP\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3rh4gxp8\\\\socketio_697229108dcc4cf39a5262e114ca36cd\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\HP\\AppData\\Local\\Temp\\pip-pip-egg-info-mhts4519'\n",
      "         cwd: C:\\Users\\HP\\AppData\\Local\\Temp\\pip-install-3rh4gxp8\\socketio_697229108dcc4cf39a5262e114ca36cd\\\n",
      "    Complete output (11 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 12, in <module>\n",
      "        from setuptools.extension import Extension\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\extension.py\", line 7, in <module>\n",
      "        from setuptools.dist import _get_unpatched\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\setuptools\\dist.py\", line 16, in <module>\n",
      "        import pkg_resources\n",
      "      File \"E:\\anaconda3\\lib\\site-packages\\pkg_resources.py\", line 1479, in <module>\n",
      "        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)\n",
      "    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ac/a1/b26c0a622dcd5f606fd384927a15441af7c4dd1cbc6258e082d7eaee8122/socketio-0.0.1.tar.gz#sha256=de909e4c94c26fe7623eb0cb3ef02721a444882e47198798851ac85d44212f4b (from https://pypi.org/simple/socketio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "ERROR: Could not find a version that satisfies the requirement netifaces==0.10.6 (from socketio)\n",
      "ERROR: No matching distribution found for netifaces==0.10.6\n"
     ]
    }
   ],
   "source": [
    "pip install socketio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "538977df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting eventlet\n",
      "  Using cached eventlet-0.33.1-py2.py3-none-any.whl (226 kB)\n",
      "Collecting dnspython>=1.15.0\n",
      "  Using cached dnspython-2.2.1-py3-none-any.whl (269 kB)\n",
      "Requirement already satisfied: greenlet>=0.3 in e:\\anaconda3\\lib\\site-packages (from eventlet) (1.0.0)\n",
      "Requirement already satisfied: six>=1.10.0 in e:\\anaconda3\\lib\\site-packages (from eventlet) (1.15.0)\n",
      "Installing collected packages: dnspython, eventlet\n",
      "Successfully installed dnspython-2.2.1 eventlet-0.33.1\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install eventlet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "abae550b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: flask in e:\\anaconda3\\lib\\site-packages (1.1.2)\n",
      "Requirement already satisfied: itsdangerous>=0.24 in e:\\anaconda3\\lib\\site-packages (from flask) (1.1.0)\n",
      "Requirement already satisfied: click>=5.1 in e:\\anaconda3\\lib\\site-packages (from flask) (7.1.2)\n",
      "Requirement already satisfied: Werkzeug>=0.15 in e:\\anaconda3\\lib\\site-packages (from flask) (1.0.1)\n",
      "Requirement already satisfied: Jinja2>=2.10.1 in e:\\anaconda3\\lib\\site-packages (from flask) (2.11.3)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in e:\\anaconda3\\lib\\site-packages (from Jinja2>=2.10.1->flask) (1.1.1)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install flask"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c769762",
   "metadata": {},
   "outputs": [],
   "source": [
    "import socketio\n",
    "import eventlet\n",
    "import numpy as np\n",
    "from flask import Flask\n",
    "from keras.models import load_model\n",
    "import base64\n",
    "from io import BytesIO\n",
    "from PIL import Image\n",
    "import cv2\n",
    "\n",
    "sio = socketio.Server()\n",
    "\n",
    "app = Flask(__name__) #'__main__'\n",
    "speed_limit = 10\n",
    "def img_preprocess(img):\n",
    "    img = img[60:135,:,:]\n",
    "    img = cv2.cvtColor(img, cv2.COLOR_RGB2YUV)\n",
    "    img = cv2.GaussianBlur(img,  (3, 3), 0)\n",
    "    img = cv2.resize(img, (200, 66))\n",
    "    img = img/255\n",
    "    return img\n",
    "\n",
    "\n",
    "@sio.on('telemetry')\n",
    "def telemetry(sid, data):\n",
    "    speed = float(data['speed'])\n",
    "    image = Image.open(BytesIO(base64.b64decode(data['image'])))\n",
    "    image = np.asarray(image)\n",
    "    image = img_preprocess(image)\n",
    "    image = np.array([image])\n",
    "    steering_angle = float(model.predict(image))\n",
    "    throttle = 1.0 - speed/speed_limit\n",
    "    print('{} {} {}'.format(steering_angle, throttle, speed))\n",
    "    send_control(steering_angle, throttle)\n",
    "\n",
    "\n",
    "\n",
    "@sio.on('connect')\n",
    "def connect(sid, environ):\n",
    "    print('Connected')\n",
    "    send_control(0, 0)\n",
    "\n",
    "def send_control(steering_angle, throttle):\n",
    "    sio.emit('steer', data = {\n",
    "        'steering_angle': steering_angle.__str__(),\n",
    "        'throttle': throttle.__str__()\n",
    "    })\n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    model = load_model('model.h5')\n",
    "    app = socketio.Middleware(sio, app)\n",
    "    eventlet.wsgi.server(eventlet.listen(('', 4567)), app)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
